demos/link-prediction/homogeneous-comparison-link-prediction.ipynb

```diff
@@ -0,0 +1,1515 @@
+{
```

Comment:
<p>Could you file an issue about expanding GCN to work with UnsupervisedSampler directly, so that we can eventually remove the loop in this cell?</p><p> </p><p>Also, why does this use the <code>LinkEmbedding</code>  class while the other models use <code>link_classification</code>  instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1658/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Reply:
<p>I will create an issue to make GCN able to work with UnsupervisedSampler.</p><p> </p><p>Here, I use <code>LinkEmbedding</code>  rather than <code>link_classification</code> for the reason that <code>link_classification</code>  cannot output the prediction with correct shape. The input <code>x_out</code>  has a different shape, compared with other embedding learning functions.</p><p> </p>